File: back_end/java_version/anti_patterns/src/main/java/com/shipay/challenge/endpoints/registration/RegistrationController.java
package com.shipay.challenge.endpoints.registration;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import javax.servlet.http.HttpServletRequest;

@RestController
public class RegistrationController {

    @Autowired
    private RegistrationManager manager;

    @GetMapping("/get-customer")
    public Customer getCustomer(HttpServletRequest request) {
        return manager.findCustomer(request);
    }
}

File: back_end/teahupoo/anti_patterns/Endpoints/Registration/RegistrationController.cs
using Microsoft.AspNetCore.Mvc;

namespace Shipay.Challenge.Endpoints.Registration
{
    [ApiController]
    [Route("api")]
    public class RegistrationController : ControllerBase
    {
        private readonly RegistrationManager _manager;

        public RegistrationController(RegistrationManager manager)
        {
            _manager = manager;
        }

        [HttpGet("get-customer")]
        public Customer GetCustomer()
        {
            var idStr = HttpContext.Request.Query["id"];
            return _manager.FindCustomer(idStr);
        }
    }
}
